src/quartics/tenten_qs.py

```diff
@@ -1,2 +1,656 @@
 '''An implementation of algorithm 1010 for solving quartic equations
+
+Algorithm 1010:
+Alberto Giacomo Orellana and Cristiano De Michele. 2020. Algorithm 1010: Boosting Efficiency in Solving
+Quartic Equations with No Compromise in Accuracy. ACM Trans. Math. Softw. 46, 2, Article 20 (May 2020),
+28 pages.
+https://doi.org/10.1145/3386241
+
+Written referencing OpenMC implementation: https://github.com/openmc-dev/openmc/blob/develop/src/external/quartic_solver.cpp
+
+Full license information described at end of file.
+'''
+
+from collections.abc import Iterable
+import sys
+import mpmath
+from mpmath import mpmathify, mpf, mpc
+from mpmath import sqrt, fabs as abs, power as pow, sign, chop
+from math import isclose
+import math
+
+
+MpfAble: type = float|int|str|mpf
+
+
+# cbrt(MAX_DOUBLE) / 1.618034
+CUBIC_RESCAL_FACT = 3.488062113727083e+102
+# pow(MAX_DOUBLE, 0.25) / 1.618034
+QUART_RESCAL_FACT = 7.156344627944542e+76
+MACHEPS = sys.float_info.epsilon
+
+
+class Solve1010:
```

Comment:
Classes should be nouns. `Algorithm1010Solver`, `Alg1010Solver`, or `TentenSolver` would be fine. Make the module name consistent. If example if you go with `Alg1010Solver` (my preference), rename this file `alg1010.py` or `alg1010_qs.py`.